src/compiler/utilities.ts

```diff
@@ -7293,57 +7305,236 @@ function Signature(this: Signature, checker: TypeChecker, flags: SignatureFlags)
     }
 }
 
-function Node(this: Mutable<Node>, kind: SyntaxKind, pos: number, end: number) {
-    this.pos = pos;
-    this.end = end;
-    this.kind = kind;
-    this.id = 0;
-    this.flags = NodeFlags.None;
-    this.modifierFlagsCache = ModifierFlags.None;
-    this.transformFlags = TransformFlags.None;
-    this.parent = undefined!;
-    this.original = undefined;
-    this.emitNode = undefined;
-}
-
-function Token(this: Mutable<Node>, kind: SyntaxKind, pos: number, end: number) {
-    this.pos = pos;
-    this.end = end;
-    this.kind = kind;
-    this.id = 0;
-    this.flags = NodeFlags.None;
-    this.transformFlags = TransformFlags.None;
-    this.parent = undefined!;
-    this.emitNode = undefined;
-}
-
-function Identifier(this: Mutable<Node>, kind: SyntaxKind, pos: number, end: number) {
-    this.pos = pos;
-    this.end = end;
-    this.kind = kind;
-    this.id = 0;
-    this.flags = NodeFlags.None;
-    this.transformFlags = TransformFlags.None;
-    this.parent = undefined!;
-    this.original = undefined;
-    this.emitNode = undefined;
-    (this as Identifier).flowNode = undefined;
-}
-
 function SourceMapSource(this: SourceMapSource, fileName: string, text: string, skipTrivia?: (pos: number) => number) {
     this.fileName = fileName;
     this.text = text;
     this.skipTrivia = skipTrivia || (pos => pos);
 }
 
+const Node = class implements Node {
+    declare kind: SyntaxKind;
+    declare flags: NodeFlags;
+    declare modifierFlagsCache: ModifierFlags;
+    declare transformFlags: TransformFlags;
+    declare id?: number | undefined;
+    declare parent: Node;

```

Comment:
Maybe I'm missing something, but why do these use `declare` as opposed to just writing `parent: Node` or `parent: Node = undefined!`?

Reply:
This is temporary. The `objectAllocator` in tsc doesn't use separate constructors for `PrivateIdentifier` and `SourceFile`, so I just want to establish a rough baseline for that minimal change first.